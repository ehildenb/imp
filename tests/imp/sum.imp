// n |-> 0  s |-> 55

int n, s;
n = 10;
s = 0;
while (! (n <= 0) ) {
    s = s + n;
    n = n - 1;
}
