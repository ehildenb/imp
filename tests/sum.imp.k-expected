<imp>
  <k>
    .
  </k>
  <mem>
    n |-> 0
    s |-> 55
  </mem>
</imp>
