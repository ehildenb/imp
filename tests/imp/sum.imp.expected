<imp>
  <k>
    .
  </k>
  <mem>
    n |-> 0
    s |-> 55
  </mem>
  <procs>
    .Map
  </procs>
  <callStack>
    .List
  </callStack>
</imp>
